WordPress/Classes/ViewRelated/Me/App Settings/Boolean User Defaults/BooleanUserDefaultsDebugViewModel.swift

```diff
@@ -0,0 +1,104 @@
+import SwiftUI
+
+typealias BooleanUserDefaultsSections = [String: BooleanUserDefaultEntries]
+typealias BooleanUserDefaultEntries = [String: BooleanUserDefault]
+
+final class BooleanUserDefaultsDebugViewModel: ObservableObject {
+    @Published private var allUserDefaultsSections = BooleanUserDefaultsSections()
+    @Published var searchQuery: String = ""
+
+    private var persistentRepository: UserPersistentRepository
+
+    var userDefaultsSections: BooleanUserDefaultsSections {
+        return filterUserDefaults(by: searchQuery)
+    }
```

Comment:
[Suggestion] Ideally, this should be `@Published`, because it is the list's source of truth, not `allUserDefaultsSections`. The goal is that whenever  `userDefaultsSections` changes, the UI should change as well.

Reply:
In my opinion, `userDefaultsSections` is not the source of truth because it's not a variable that stores data - it's a wrapper over `allUserDefaultsSections` that filters only those sections that should be visible accordingly to the search query. `userDefaultsSections` value cannot be changed because it doesn't store any value but fully depends on `allUserDefaultsSections` and `searchQuery` which _are_ the source of truth.
There is no way to set a value directly to `userDefaultsSections`, we have to change `allUserDefaultsSections`  instead. This is the reason why I used `@Published` for the top level data.

Reply:
Discussed it on Slack and decided that `@Published` has to be used for renderable data only, but for tracking `allUserDefaultsSections` changes we are going to use `didSet` modifier.